Misc/Mode.java

```diff
@@ -0,0 +1,96 @@
+package Misc;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * This class is used to retrieve all the modes of a given array. Mode is the value that appears the
+ * most in a given set. If there are more than one mode then they are returned sorted from min to
+ * max.
+ *
+ * @author Kostas
+ * @param <E> The type of the array
+ */
+public class Mode<E extends Comparable<? super E>> {
+
+  public static void main(String args[]) {
+    String[] values = {"Bravo", "Alpha", "Mega", "Alpha", "Bravo", "Mega"};
+    Mode<String> md = new Mode<>();
+    ArrayList<String> modes = md.getMode(values);
+    System.out.println(modes.toString());
+  }
+
+  /**
+   * This method is used to extract all modes of the given array sorted in an ArrayList.
+   *
+   * @param array to extract the modes from
+   * @return a sorted {@link ArrayList} with all the modes
+   */
+  public ArrayList<E> getMode(E[] array) {
+
+    // initializing a Map where the key is an element of the array and the value is the element's
+    // appearances
+    HashMap<E, Integer> appCounter = new HashMap<>();
```

Comment:
It could be better if you change the name of hashMap to something else like freqCounter, so it clearly indicates it calculates frequency.